packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -914,27 +915,19 @@ class HotRunner extends ResidentRunner {
     // Record time it took for the VM to reload the sources.
     _addBenchmarkData('hotReloadVMReloadMilliseconds', vmReloadTimer.elapsed.inMilliseconds);
     final Stopwatch reassembleTimer = Stopwatch()..start();
-    // Reload the isolate.
-    final List<Future<void>> allDevices = <Future<void>>[];
-    for (final FlutterDevice device in flutterDevices) {
-      globals.printTrace('Sending reload events to ${device.device.name}');
-      final List<Future<ServiceObject>> futuresViews = <Future<ServiceObject>>[];
-      for (final FlutterView view in device.views) {
-        globals.printTrace('Sending reload event to "${view.uiIsolate.name}"');
-        futuresViews.add(view.uiIsolate.reload());
-      }
-      allDevices.add(Future.wait(futuresViews).whenComplete(() {
-        return device.refreshViews();
-      }));
-    }
-    await Future.wait(allDevices);
+
+    // Reload the isolate data.
+    await Future.wait(<Future<void>>[
+      for (final FlutterDevice device in flutterDevices)
+        device.refreshViews()
```

Comment:
Is `device.refreshViews()` now also doing the `uiIsolate.reload()`?

Reply:
Yeah, we only need the isolateRef data so reloading the full isolate isn't necessary (and we do this as needed anyway below)